packages/integrations/solid/src/context.ts

```diff
@@ -11,10 +11,13 @@ export function getContext(result: RendererContext['result']): Context {
 	if (contexts.has(result)) {
 		return contexts.get(result)!;
 	}
-	let ctx = {
+	let ctx: Context = {
 		c: 0,
 		get id() {
-			return 's' + this.c.toString();
+			// a hyphen at the end makes it easier to distinguish the island
+			// render id from component tree portion of the hydration key `data-hk="..."`
+			// https://github.com/solidjs/solid-start/blob/f0860887030e0632949b3f497e279aecb6ed5afd/packages/start/islands/mount.tsx#L41
```

Comment:
I'll just remove it from the PR because I don't want the PR to get distracted by this. I changed this because it made debugging hydration errors a little bit easier and forgot about it, but now that you mention it, I realize that technically it's not related or essential. Maybe me or someone else can make a PR for this some other time. :)